### PR TITLE
Add all available options to example config

### PR DESCRIPTION
**Pull request by Perflyst**

### CODE REVIEW

File: config/config.example.yml [hunk 1]
```diff
@@ -1,11 +1,147 @@
...
+# Define if accessed via HTTPS
 https_only: false
-domain:
+
+# Signing key for CSRF tokens
+# randomly generated by default
+#hmac_key: 
+
+# External domain
+domain: replace.me
```
Comment (TheFrenchGhosty) [inline config/config.example.yml:41]: Should be empty for users running it locally.

File: config/config.example.yml [hunk 2]
```diff
@@ -1,11 +1,147 @@
...
+
+# User login
+login_enabled: true
+
+# User registration
+registration_enabled: true
+
+# Instance statistics
+# Available on /api/v1/stats
+statistics_enabled: false
```
Comment (TheFrenchGhosty) [inline config/config.example.yml:62]: More information should be added to explain what this shows, along with mentioning that it's useful to have for instances.invidio.us (and should be therefore enabled for public instances).

File: config/config.example.yml [hunk 3]
```diff
@@ -1,11 +1,147 @@
...
+#external_port: 
+
+# Default preferences for
+# new and unregistered users
+default_user_preferences:
+  annotations: false
+  annotations_subscribed: false
+  autoplay: false
+  captions:
+  - ""
```
Comment (TheFrenchGhosty) [inline config/config.example.yml:79]: ```dash_quality``` should be added.

File: config/config.example.yml [hunk 4]
```diff
@@ -1,11 +1,147 @@
...
+  - ""
+  - ""
+  comments:
+  - youtube
+  - ""
```
Comment (TheFrenchGhosty) [inline config/config.example.yml:84]: Reddit comments are enabled by default.

File: config/config.example.yml [hunk 5]
```diff
@@ -1,11 +1,147 @@
...
+  continue: false
+  continue_autoplay: true
+  dark_mode: ""
```
Comment (TheFrenchGhosty) [inline config/config.example.yml:87]: Set to false by default.

File: config/config.example.yml [hunk 6]
```diff
@@ -1,11 +1,147 @@
...
+# HTTP host for internal webserver
+host_binding: 0.0.0.0
+
+# Pool for HTTP connections to YouTube
+pool_size: 100
+
+# Cookies for connection to YouTUbe
+#cookies: 
+
+# Anti-Captcha configuration for reCaptcha
```
Comment (TheFrenchGhosty) [inline config/config.example.yml:144]: Should mention that both anti-captcha, and captcha-monsters are available as provider.

File: config/config.example.yml [hunk 7]
```diff
@@ -1,11 +1,147 @@
...
+# Available on /api/v1/stats
+statistics_enabled: false
+
+# Instance admin account
+# account needs to exists
+admins: []
+
+# If running behind reverse proxy
+# an external port is required
+#external_port:
```
Comment (TheFrenchGhosty) [inline config/config.example.yml:70]: Is 3000 by default, and should mention that this is critical.

**Review by TheFrenchGhosty — CHANGES_REQUESTED** [review]
Why did you set some settings to their default value, but commented out some other settings, in my opinion commenting out the "optional" stuff, and keeping everything important uncommented would be better.

File: config/config.example.yml [hunk 8]
```diff
@@ -1,11 +1,147 @@
...
+# randomly generated by default
+#hmac_key: 
+
+# External domain
+domain: replace.me
+
+# Use server-side notifications provided by YouTube
+# https://developers.google.com/youtube/v3/guides/push_notifications
+# Requires domain and hmac_key to be set
+use_pubsub_feeds: false
```
Comment (saltycrys) [inline config/config.example.yml:46]: This does not require `hmac_key` to be set.

Might be good to point out that `domain: localhost` works, too.

File: config/config.example.yml (hunk 1 above)
Comment (saltycrys) [reply config/config.example.yml:41]: Might be good to point out that storyboards need this to be set and for local (non Docker) instances `domain: localhost` works.

File: config/config.example.yml (hunk 8 above)
Comment (saltycrys) [reply config/config.example.yml:46]: Also setting this to a number higher than zero sets the number of "threads" (fibers really) that are used for this. See https://github.com/iv-org/invidious/blob/168376b04671c6e362468267a3b07dd332b40702/src/invidious/jobs/subscribe_to_feeds_job.cr#L11-L13

There's a bug right now where setting this to `false` doesn't disable it, I'll fix that shortly.

File: config/config.example.yml (hunk 1 above)
Comment (TheFrenchGhosty) [reply config/config.example.yml:41]: Then I guess it should be set to localhost.

Comment (saltycrys) [reply config/config.example.yml:41]: I don't think that works for Docker.

**Pull request closed without merging**